enterprise/backend/src/metabase_enterprise/sandbox/api/util.clj

```diff
@@ -34,9 +35,10 @@
     (filter (partial enforce-sandbox? group-id->perms-set)
             sandboxes)))
 
-(defn segmented-user?
+(defenterprise segmented-user?
   "Returns true if the currently logged in user has segmented permissions. Throws an exception if no current user
   is bound."
+  :feature :sandboxes
   []
   (boolean
    (when-not *is-superuser?*
```

Comment:
just thinking out loud here: I wonder whether we can cache this smartly. Search perf is already bad and this makes it even worse for enterprise users.

Reply:
we could maybe improve the performance a bit by combine the 2 selects into one?

Reply:
not necessary to be in this PR. lmk if you want me to open a PR for this.

Reply:
i'm open to that improvement, but this only impacts enterprise users with the sandbox feature, whereas previously it would impact anyone with the enterprise namespaces on the classpath. It's true search is slow but that's because it does lots of full table scans. These extra app db queries don't insult me at the moment. But I'm fine with a follow-on issue to speed this up.